frontend/src/app/tasks/components/SaveTemplateModal.tsx

```diff
@@ -1,64 +1,74 @@
+import { useState } from 'react'
 import { Sheet, Typography, Modal, ModalClose } from '@mui/joy'
-
 import { Input } from '@/ui/Input/Input'
 import { Task } from '@/domain/Task'
 import { createTemplate } from '../actions/createTemplate'
 import { useAlert } from '@/ui/Alert/useAlert'
 import { SubmitButton } from '@/ui/SubmitButton/SubmitButton'
+import { SaveCopy24Filled } from '@fluentui/react-icons'
+import { IconButton } from './IconButton'
 
 type SaveAsTemplateProps = {
-  closeModal: () => void
-  open: boolean
   task: Task
 }
 
-export const SaveTemplateModal = ({ closeModal, open, task }: SaveAsTemplateProps) => {
+export const SaveTemplateModal = ({ task }: SaveAsTemplateProps) => {
   const { showError, showSuccess } = useAlert()
+  const [templateModalOpen, setTemplateModalOpen] = useState(false)
+
+  const closeModal = () => setTemplateModalOpen(false)
 
   return (
-    <Modal
-      sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}
-      open={open}
-      onClose={() => closeModal()}
-      aria-labelledby="confirmation-modal-title"
-      aria-describedby="modal-content"
-    >
-      <Sheet
-        variant="outlined"
-        component="form"
-        action={async (formData: FormData) => {
-          const template = await createTemplate(task, formData)
-          if (template.error) {
-            showError(template.error)
-          } else {
-            showSuccess('Template created')
-          }
-          closeModal()
-        }}
-        sx={{
-          minWidth: 300,
-          minHeight: 200,
-          borderRadius: 'md',
-          padding: 3,
-          boxShadow: 'lg',
-          display: 'flex',
-          flexDirection: 'column',
-          gap: '8px'
-        }}
+    <>
+      <IconButton
+        onClick={closeModal}
+        icon={SaveCopy24Filled}
+        description="save task as template"
+      />
+      <Modal
+        sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}
+        open={templateModalOpen}
+        onClose={closeModal}
+        aria-labelledby="confirmation-modal-title"
+        aria-describedby="modal-content"
```

Comment:
There is no element with the id of `modal-content` so this will toss an a11y  error.

Reply:
I removed that because I found no use for it :thinking: there's no need to describe the modal since we already associate it to the title